docs/source/ruletypes.rst

```diff
@@ -153,67 +153,111 @@ Rule Configuration Cheat Sheet
+|                                                       |        |           |           |        |           |       |          |        |           |                  |                 |                |
+|``query_key`` (string or list, no default)             |        |           |           |        |           |       |          |        |           |                  |                 |                |
+|                                                       |        |           |           |        |           |       |          |        |           |                  |                 |                |
+|``terms_size`` (int, default 50)                       |        |           |           |        |           |       |          |        |           |                  |                 |                |
++-------------------------------------------------------+--------+-----------+-----------+--------+-----------+-------+----------+--------+-----------+------------------+-----------------+----------------+
+| ``spike_height`` (int, no default)                    |        |           |           |        |           |   Req |          |        |           |                  |  Req            |                |
++-------------------------------------------------------+--------+-----------+-----------+--------+-----------+-------+----------+--------+-----------+------------------+-----------------+----------------+
+|``spike_type`` ([up|down|both], no default)            |        |           |           |        |           |   Req |          |        |           |                  |  Req            |                |
++-------------------------------------------------------+--------+-----------+-----------+--------+-----------+-------+----------+--------+-----------+------------------+-----------------+----------------+
+|``alert_on_new_data`` (boolean, default False)         |        |           |           |        |           |   Opt |          |        |           |                  |                 |                |
++-------------------------------------------------------+--------+-----------+-----------+--------+-----------+-------+----------+--------+-----------+------------------+-----------------+----------------+
+|``threshold_ref`` (int, no default)                    |        |           |           |        |           |   Opt |          |        |           |                  |                 |                |
++-------------------------------------------------------+--------+-----------+-----------+--------+-----------+-------+----------+--------+-----------+------------------+-----------------+----------------+
+|``threshold_ref`` (number, no default)                 |        |           |           |        |           |       |          |        |           |                  |  Opt            |                |
```

Comment:
Did you intend to duplicate the threshold_ref and threshold_cur values? One is labeled as a number, the other as an int.

Reply:
look  schema.yaml

Spike
https://github.com/jertel/elastalert2/blob/master/elastalert/schema.yaml#L131
```yaml
      threshold_ref: {type: integer}
      threshold_cur: {type: integer}
```

Spike Aggregation
https://github.com/jertel/elastalert2/blob/master/elastalert/schema.yaml#L147
```yaml
      threshold_ref: {type: number}
      threshold_cur: {type: number}
```

Reply:
The difference between number and integer is that number is allowed after the decimal point. integer is an integer only

Reply:
It seems that such schema.yaml is set for the above reason.